openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java

```diff
@@ -1439,8 +1439,6 @@ && userHasRelationshipWithAnyBot(original, bot)) {
               create.getAuthenticationMechanism(),
               original.getAuthenticationMechanism());
       user.setRoles(original.getRoles());
-    } else if (bot != null && ProviderType.SYSTEM.equals(bot.getProvider())) {
-      user.setRoles(UserUtil.getRoleForBot(botName));
```

Comment:
I could not figure out a way to get inside this path. In any case, train of thought is that if there is already a user, we won't re-apply the bot roles